src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -140,8 +137,16 @@ public Map<Integer, Float> searchLeaf(LeafReaderContext context, int k) throws I
          * This improves the recall.
          */
         Map<Integer, Float> docIdsToScoreMap;
+        final ExactSearcher.ExactSearcherContext exactSearcherContext = ExactSearcher.ExactSearcherContext.builder()
+            .k(k)
+            .isParentHits(true)
+            .matchedDocs(filterBitSet)
+            .leafReaderContext(context)
+            .useQuantizedVectorsForSearch(true)
```

Comment:
Same for this, add a comment of why its kept true. Is there a way we can pass in QuantizationConfig here? its available in mappedFieldType

Reply:
Its not about passing the Quantized config. Even when Quantized Config is present, we might not want to do a quantization based exact search. So every caller knows on what type of vectors it wants to do exact search.

Reply:
So the idea is to make the decision early with quantization config (if its possible). In this case, we are adding true as default and then falling back on quantization params as null. Checking quantization params checks for quantization config. anyways not a blocker as its not expensive if its not present

Reply:
> So the idea is to make the decision early with quantization config (if its possible)

Here I took the other approach. The flow tells exact search to do search on Quantized vectors or not. This has some benefits: 
1. The logic of what vectors to use for search is defined by the flow and not dependent on Quantization param. Biggest reason for that is : for rescoring if you look at we have the Quantization params but we want to search via full precision vectors. 

If we move towards the use of QuantizationParams being null or not in fieldInfo to say what vectors to use for quantization then it will not work for rescoring.